src/cmake/modules/CUnit/src/main.c.in

```diff
@@ -9,23 +9,224 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
  */
-#include "CUnit/Runner.h"
+#include <stdio.h>
+#include <string.h>
 
-@CUnit_Decls@
+#include <CUnit/Basic.h>
+#include <CUnit/Automated.h>
+#include <CUnit/Test.h>
 
-int main (int argc, char *argv[])
+static struct {
+    bool print_help;
+    bool automated;
+    bool junit;
+    const char *results;
+    CU_BasicRunMode mode;
+    CU_ErrorAction error_action;
+    const char *suite;
+    const char *test;
+} opts = {
+    false,
+    false,
+    false,
+    NULL,
+    CU_BRM_NORMAL,
+    CUEA_IGNORE,
+    "*",
+    "*"
+};
+
+int patmatch(const char *pat, const char *str)
+{
+    while (*pat) {
+        if (*pat == '?') {
+            /* any character will do */
+            if (*str++ == 0) {
+                return 0;
+            }
+            pat++;
+        } else if (*pat == '*') {
+            /* collapse a sequence of wildcards, requiring as many
+               characters in str as there are ?s in the sequence */
+            while (*pat == '*' || *pat == '?') {
+                if (*pat == '?' && *str++ == 0) {
+                    return 0;
+                }
+                pat++;
+            }
+            /* try matching on all positions where str matches pat */
+            while (*str) {
+                if (*str == *pat && patmatch(pat+1, str+1)) {
+                    return 1;
+                }
+                str++;
+            }
+            return *pat == 0;
+        } else {
+            /* only an exact match */
+            if (*str++ != *pat++) {
+                return 0;
+            }
+        }
+    }
+
+    return *str == 0;
+}
+
+static void usage(const char *name)
+{
+    fprintf(stderr, "Usage: %s OPTIONS\n", name);
+    fprintf(stderr, "Try '%s -h' for more information\n", name);
+}
+
+static void help(const char *name)
+{
+    printf("Usage: %s [OPTIONS]\n", name);
+    printf("\n");
+    printf("Possible options:\n");
+    printf("  -a           run in automated mode\n");
+    printf("  -f           fail fast\n");
+    printf("  -h           display this help and exit\n");
+    printf("  -j           junit format results \n");
+    printf("  -r FILENAME  results file for automated run\n");
+    printf("  -s PATTERN   run only tests in suites matching pattern\n");
+    printf("  -t PATTERN   run only test matching pattern\n");
+}
+
+static CU_ErrorCode parse_options(int argc, char *argv[])
```

Comment:
I would prefer that ``parse_options`` returns 0 on success and a non-zero ``int`` that happens to also be the exit status of the program on failure - 0 on success, 1 or 2 or so on failure. That avoids abuse of ``CU_ErrorCode`` like in the default case